aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -1931,29 +1928,60 @@ void scaled_gemm(
+    TORCH_CHECK(false, "scaled_gemm with `torch.float8_e4m3fn` scales of 1x16 blocks is only supported for CUDA 12.8 and above");
 #endif // if CUDA_VERSION >= 12080
-  } else if (mat1_scale_dtype == kFloat && mat2_scale_dtype == kFloat && use_rowwise) {
+  } else if (mat1_scaling_type == ScalingType::RowWise && mat2_scaling_type == ScalingType::RowWise) {
+    TORCH_CHECK(mat1_scale_dtype == kFloat);
+    TORCH_CHECK(mat2_scale_dtype == kFloat);
 #if CUDA_VERSION >= 12090 || (defined(USE_ROCM) && defined(HIPBLASLT_OUTER_VEC))
     computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_A_SCALE_MODE, CUBLASLT_MATMUL_MATRIX_SCALE_OUTER_VEC_32F);
     computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_B_SCALE_MODE, CUBLASLT_MATMUL_MATRIX_SCALE_OUTER_VEC_32F);
 #elif defined(USE_ROCM) && defined(HIPBLASLT_VEC_EXT)
     // no-op here for older hipblaslt ext enums, to avoid TORCH_CHECK below
 #else
-    TORCH_CHECK(false, "scaled_gemm with `torch.float` outer vector scaling is only supported for CUDA 12.9 and above");
+    TORCH_CHECK(false, "scaled_gemm with rowwise scaling is only supported for CUDA 12.9 and above");
 #endif // if CUDA_VERSION >= 12090
+  } else if ((mat1_scaling_type == ScalingType::BlockWise1x128 || mat1_scaling_type == ScalingType::BlockWise128x128) && (mat2_scaling_type == ScalingType::BlockWise1x128 || mat2_scaling_type == ScalingType::BlockWise128x128)) {
+    TORCH_CHECK(mat1_scale_dtype == kFloat);
+    TORCH_CHECK(mat2_scale_dtype == kFloat);
+    TORCH_CHECK(
+      mat1_scaling_type != ScalingType::BlockWise128x128 || mat2_scaling_type != ScalingType::BlockWise128x128,
+      "scaled_gemm doesn't support both operands using 128x128 blockwise scaling");
+    TORCH_CHECK(!use_fast_accum, "scaled_gemm doesn't support fast accum with 1x128 and 128x128 blockwise scaling")
+#if CUDA_VERSION >= 12090
+    computeDesc.setAttribute(
+      CUBLASLT_MATMUL_DESC_A_SCALE_MODE,
+      mat1_scaling_type == ScalingType::BlockWise1x128 ? CUBLASLT_MATMUL_MATRIX_SCALE_VEC128_32F : CUBLASLT_MATMUL_MATRIX_SCALE_BLK128x128_32F);
+    computeDesc.setAttribute(
+      CUBLASLT_MATMUL_DESC_B_SCALE_MODE,
+      mat2_scaling_type == ScalingType::BlockWise1x128 ? CUBLASLT_MATMUL_MATRIX_SCALE_VEC128_32F : CUBLASLT_MATMUL_MATRIX_SCALE_BLK128x128_32F);
+#else
+    TORCH_CHECK(false, "scaled_gemm with 1x128 and 128x128 blockwise scaling is only supported for CUDA 12.9 and above");
+#endif // if CUDA_VERSION >= 12090
+  } else if (mat1_scaling_type == ScalingType::TensorWise && mat2_scaling_type == ScalingType::TensorWise) {
+    TORCH_CHECK(mat1_scale_dtype == kFloat);
+    TORCH_CHECK(mat2_scale_dtype == kFloat);
+    // This is the default, but make it explicit.
+    computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_A_SCALE_MODE, CUBLASLT_MATMUL_MATRIX_SCALE_SCALAR_32F);
+    computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_B_SCALE_MODE, CUBLASLT_MATMUL_MATRIX_SCALE_SCALAR_32F);
+  } else {
```

Comment:
Looks good, thanks for doing this. Side note I find these macro soup of a very hard to reason about now. I was doing some changes here: https://github.com/pytorch/pytorch/pull/156458/files when I was trying to debug what was going on w/ NVFP4. 

 Not blocking for this PR but if you wanted to refactor some more to make this easier to reason about I wouldnt hate it :)